.github/workflows/build.yml

```
@@ -131,11 +131,111 @@ jobs:
       - name: Run make for java
         run: make -C lib/java
 
+      - name: Run make install for java
+        run: make -C lib/java install
+
+      - name: Upload java libthrift artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: libthrift
+          if-no-files-found: error
+          path: ~/.m2/repository/org/apache/thrift
+
       - name: Run make check for java
         run: make -C lib/java check
 
+      - name: Run make precross for java
+        run: make -C lib/java precross
+
+      - name: Upload java precross artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: java-precross
+          if-no-files-found: error
+          path: |
+            lib/java/build/functionalTestJar/
+            lib/java/build/runclient
+            lib/java/build/runnonblockingserver
+            lib/java/build/runserver
+            lib/java/build/runservletserver
+          retention-days: 3
+
       - name: Run make for kotlin
         run: make -C lib/kotlin
 
       - name: Run make check for kotlin
         run: make -C lib/kotlin check
+
+      - name: Run make precross for kotlin
+        run: make -C lib/kotlin precross
+
+      - name: Upload kotlin precross artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: kotlin-precross
+          if-no-files-found: error
+          path: |
+            lib/kotlin/cross-test-client/build/install/TestClient/
+            lib/kotlin/cross-test-server/build/install/TestServer/
+          retention-days: 3
+
+  cross-test:
+    needs:
+      - lib-java-kotlin
+    strategy:
+      matrix:
+        test_regex:
+          - "-(binary)"
+          - "-(header|json)"
+          - "-(compact|multiplexed)"
```

I wonder if the failure is because the leading dash is being treated as a CLI arg instead of part of the regex pattern?

```suggestion
          - "[-](binary)"
          - "[-](header|json)"
          - "[-](compact|multiplexed)"
```